src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -181,18 +185,18 @@ public InnerClassLambdaMetafactory(MethodHandles.Lookup caller,
         // generating bridges in the target class)
         useImplMethodHandle = (Modifier.isProtected(implInfo.getModifiers()) &&
                                !VerifyAccess.isSamePackage(targetClass, implInfo.getDeclaringClass())) ||
```

Comment:
If the `implClass` is hidden, then it’s not possible to generate bytecode which correctly refers to `implMethod`, so the only way is to use the method handle in that case:
```suggestion
        useImplMethodHandle = (Modifier.isProtected(implInfo.getModifiers()) &&
                               !VerifyAccess.isSamePackage(targetClass, implInfo.getDeclaringClass())) ||
                               implClass.isHidden() ||
```

Reply:
I think this belongs to another RFE and is out of scope of this one. In addition, we need extra tests to invoke the hidden class's lambda to assert its behavioral correctness. Currently, the previously failing test [`LambdaProxyCaller`](https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/test-classes/LambdaProxyCallerIsHiddenApp.java) only creates the hidden class for CDS testing; it doesn't actually invoke the lambda to test its implementation correctness.